Source/Components/RView.swift

```diff
@@ -0,0 +1,768 @@
+    ///   - inputName: the name of the `Number` input
+    ///   - value: the new value for the state to hold
+    open func setNumberState(_ stateMachineName: String, inputName: String, value: Float) throws {
+        let stateMachineInstances = try _getOrCreateStateMachines(animationName: stateMachineName)
+        try stateMachineInstances.forEach { stateMachine in
+            stateMachine.getNumber(inputName).setValue(value)
+            try _play(stateMachine)
+        }
+        runTimer()
+    }
+    
+    private func _getOrCreateStateMachines(animationName: String) throws -> [RiveStateMachineInstance] {
+        let stateMachineInstances = _stateMachines(withAnimationName: animationName)
+        if stateMachineInstances.isEmpty {
+            guard let guardedArtboard = artboard else { return [] }
+            
+            let stateMachineInstance = try guardedArtboard.stateMachine(fromName: animationName).instance()
+            return [stateMachineInstance]
+        }
+        return stateMachineInstances
+    }
+    
+    private func _getOrCreateLinearAnimationInstances(animationName: String) throws -> [RiveLinearAnimationInstance] {
+        let animationInstances = _animations(withName: animationName)
+        
+        if animationInstances.isEmpty {
+            guard let guardedArtboard = artboard else { return [] }
+            
+            let animationInstance = try guardedArtboard.animation(fromName: animationName).instance()
+            return [animationInstance]
+        }
+        return animationInstances
+    }
+    
+    private func _playAnimation(
+        animationName: String,
+        loop: Loop = .loopAuto,
+        direction: Direction = .directionAuto,
+        isStateMachine: Bool = false
+    ) throws {
+        if isStateMachine {
+            let stateMachineInstances = try _getOrCreateStateMachines(animationName: animationName)
+            try stateMachineInstances.forEach { try _play($0) }
+        } else {
+            let animationInstances = try _getOrCreateLinearAnimationInstances(animationName: animationName)
+            animationInstances.forEach { _play(animation: $0, loop: loop, direction: direction) }
+        }
+    }
+    
+    private func _animations(withName animationName: String) -> [RiveLinearAnimationInstance] {
+        return _animations(withNames: [animationName])
+    }
+    
+    private func _animations(withNames animationNames: [String]) -> [RiveLinearAnimationInstance] {
+        return animations.filter { animationNames.contains($0.animation().name()) }
+    }
+    
+    private func _stateMachines(withAnimationName animationName: String) -> [RiveStateMachineInstance] {
+        return _stateMachines(withAnimationNames: [animationName])
+    }
+    
+    private func _stateMachines(withAnimationNames animationNames: [String]) -> [RiveStateMachineInstance] {
+        return stateMachines.filter { animationNames.contains($0.stateMachine().name()) }
```

Comment:
I think this already exists in `RiveView` today, but just realizing, should this be `stateMachineNames.contains(...)`?

Reply:
I don't think so.. The purpose here is to return a new array of `RiveStateMachineInstance`(s) as a subset of the instance's `stateMachines` array. The `.filter{...}` returns all elements in the `stateMachines` array that pass a `Bool` check. In this case we're checking that each element has a `.stateMachine().name()` `String` that is also contained in the `animationNames` array.

Calling `stateMachineNames.contains(...)` would return a `Bool` rather than a new array.